### PR TITLE
[ADD] Relations and CRUD for character

### PR DESCRIPTION
**Pull request by Juanda0**
In this PR I create the relations and crud for the character model

### CODE REVIEW

File: app/Models/Character.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    {
+        $this->name = $name;
+    }
+
+    public function getHead(): Item
+    {
+        return $this->items()->wherePivot('type', 'head')->first();
+    }
+
+    public function setHead(Item $item)
```
Comment (Jguerra47) [inline app/Models/Character.php:47]: Return type missing

File: app/Models/Character.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            $item->id => ['type' => 'head'],
+        ]);
+    }
+
+    public function getChest(): Item
+    {
+        return $this->items()->wherePivot('type', 'chest')->first();
+    }
+
+    public function setChest(Item $item)
```
Comment (Jguerra47) [inline app/Models/Character.php:59]: Idem

File: app/Models/Character.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            $item->id => ['type' => 'chest'],
+        ]);
+    }
+
+    public function getLegs(): Item
+    {
+        return $this->items()->wherePivot('type', 'legs')->first();
+    }
+
+    public function setLegs(Item $item)
```
Comment (Jguerra47) [inline app/Models/Character.php:71]: Idem

File: app/Models/Item.php [hunk 4]
```diff
@@ -128,4 +128,9 @@ public function setReviews(Collection $reviews): void
         $this->reviews = $reviews;
     }
 
+    public function characters()
```
Comment (Jguerra47) [inline app/Models/Item.php:131]: Idem

File: database/factories/CharacterFactory.php [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Database\Factories;
+
+use App\Models\Item;
```
Comment (Jguerra47) [inline database/factories/CharacterFactory.php:5]: Item is not used in this factory

**Review by Jguerra47 — CHANGES_REQUESTED** [review]
Leave you some comments

File: routes/web.php [hunk 6]
```diff
@@ -37,4 +37,8 @@
     Route::delete('/admin/users/{id}', 'App\Http\Controllers\AdminUserController@delete')->name('admin.user.delete');
 });
 
+Route::middleware(['auth', 'user'])->group(function () {
+    Route::get('/user/{id}/character', 'App\Http\Controllers\CharacterController@debug')->name('character.debug');
```
Comment (DanielHernandezO) [inline routes/web.php:41]: Are you sure it should be there?

**Review by DanielHernandezO — APPROVED** [review]

**Review by Jguerra47 — APPROVED** [review]

**Pull request merged**